src/main/java/com/labsynch/labseer/chemclasses/bbchem/ChemStructureServiceBBChemImpl.java

```diff
@@ -609,22 +621,48 @@ public StrippedSaltDTO stripSalts(CmpdRegMolecule inputStructure) throws CmpdReg
 	public HashMap<String, CmpdRegMolecule> standardizeStructures(HashMap<String, String> structures) throws CmpdRegMolFormatException {
 
 		// Get preprocessed structures (standardized mol structures)
-		HashMap<String, String> preprocessedStructures = new HashMap<String, String>();
+		HashMap<String, Entry<String, String>>preprocessedStructures = new HashMap<String, Entry<String, String>>();
 		try {
 			preprocessedStructures = bbChemStructureService.getPreprocessedStructures(structures);
 		} catch (IOException e) {
 			logger.error("Error getting preprocessed structures: "+e.getMessage());
 			throw new CmpdRegMolFormatException("Error getting preprocessed structures: "+e.getMessage());
 		}
 
+		// Only structures which have passed preprocessing should be passed to be processed so we need to remove those with an error
+		// Loop through the preprocessStructures and remove any structures that have an error
+		HashMap<String, String> structuresToProcess = new HashMap<String, String>();
+		for (String structureKey : preprocessedStructures.keySet()){
+			Entry<String, String> preprocessedStructure = preprocessedStructures.get(structureKey);
+			if (!preprocessedStructure.getKey().equals("SUCCESS")) {
+				logger.error("Got unexpected status '"+preprocessedStructure.getKey()+"' when preprocessing structure for id:'"+structureKey+"' molstruture: '" + structures.get(structureKey) + "'");
+				// Just add the original structure to the list of structures to process
+				// We will add the error message to the structure later
+				// TODO: Make sure the statement ends up true before merging this code
```

Comment:
I think you're doing this down on line 660, right? So can probably remove this TODO

Reply:
Yes, thanks forgot to remove this after fixing that.